test/inputs-dovetail-bytes-test.js

```diff
@@ -0,0 +1,88 @@
+'use strict';
+
+const support = require('./support');
+const bigquery = require('../lib/bigquery');
+const DovetailBytes = require('../lib/inputs/dovetail-bytes');
+
+describe('dovetail-bytes', () => {
+
+  let bytes = new DovetailBytes();
+
+  it('recognizes bytes records', () => {
+    expect(bytes.check({})).to.be.false;
+    expect(bytes.check({type: 'anything'})).to.be.false;
+    expect(bytes.check({type: 'bytes'})).to.be.true;
+    expect(bytes.check({type: 'segmentbytes'})).to.be.true;
+  });
+
+  it('formats download byte inserts', () => {
+    const record = bytes.formatDownload({
+      timestamp: 1490827132999,
+      request_uuid: 'the-uuid',
+      bytes_downloaded: 123,
+      seconds_downloaded: 1.23,
+      percent_downloaded: 0.12,
+    })
+    expect(record).to.have.keys('insertId', 'json');
+    expect(record.insertId).to.equal('bytes-the-uuid');
+    expect(record.json).to.have.keys('timestamp', 'request_uuid', 'bytes', 'seconds', 'percent');
+    expect(record.json.timestamp).to.equal(1490827132);
+    expect(record.json.request_uuid).to.equal('the-uuid');
+    expect(record.json.bytes).to.equal(123);
+    expect(record.json.seconds).to.equal(1.23);
+    expect(record.json.percent).to.equal(0.12);
+  });
+
+  it('formats impression byte inserts', () => {
```

Comment:
I'm a little unsure about the differences between `download`/`impression`/`bytes`/`segmentbytes` record types. Or maybe what those record types mean. Is there a good reference for that?

I sort of get the idea they related to how the podcast User Agent interacts with dovetail stitch artifacts?

Reply:
Reviewing this is helpful: https://github.com/PRX/dovetail-counts-lambda/blob/10bebc8415a7a9908c4e8f0eedf7b930c8603d39/README.md#kinesis-impressions-stream

Reply:
Yeah, so the `bytes` and `segmentbytes` record come from that counts lambda.  And the other 2 come from [ECS dovetail](https://github.com/PRX/dovetail.prx.org/blob/master/lib/util/impression/index.js#L62), and are non-IAB compliant, since they're logged before the actual download from the CDN.